diff -ur Argyll_V1.3.2.old/Jamfile Argyll_V1.3.2/Jamfile
--- Argyll_V1.3.2.old/Jamfile	2010-11-03 05:18:04.000000000 -0400
+++ Argyll_V1.3.2/Jamfile	2011-01-03 17:37:41.644817066 -0500
@@ -35,17 +35,6 @@
 	SubInclude tiff ;
 }
 
-if $(USE_LIBUSB1) = true {
-	SubInclude libusb1 ;
-} else {
-	if $(NT) {
-		SubInclude libusbw ;
-	}
-	if $(UNIX) {
-		SubInclude libusb ;
-	}
-}
-
 if $(UNIX) && $(OS) != MACOSX {
 	SubInclude jcnf ;
 	SubInclude ucmm ;
diff -ur Argyll_V1.3.2.old/Jamtop Argyll_V1.3.2/Jamtop
--- Argyll_V1.3.2.old/Jamtop	2010-11-03 05:18:04.000000000 -0400
+++ Argyll_V1.3.2/Jamtop	2011-01-03 17:36:29.072694056 -0500
@@ -41,7 +41,7 @@
 		} else {
 			EXIT Unable to locate the X11 include files ;
 		}
-		LINKFLAGS += -L$(LibWinD) -ldl -lrt -lX11 -lXext -lXxf86vm -lXinerama -lXrandr -lXau -lXdmcp -lXss ;
+		LINKFLAGS += -L$(LibWinD) -ldl -lrt -lX11 -lXext -lXxf86vm -lXinerama -lXrandr -lXau -lXdmcp -lXss -lusb ;
 		# HDRS += $(LibWinH) ; 
 	}
 }
diff -ur Argyll_V1.3.2.old/jcnf/jcnf.c Argyll_V1.3.2/jcnf/jcnf.c
--- Argyll_V1.3.2.old/jcnf/jcnf.c	2010-11-03 05:19:31.000000000 -0400
+++ Argyll_V1.3.2/jcnf/jcnf.c	2011-01-12 16:30:59.370109716 -0500
@@ -726,18 +726,22 @@
 static jc_error jcnf_lock_file(jcnf *p) {
 #ifndef NT
 	int i, fh;
-	int lop;
+	struct flock lock;
 
 	fh = fileno(p->fp);
 
+	lock.l_start = 0;
+	lock.l_whence = SEEK_SET;
+	lock.l_len = 0;
+
 	if (p->modify) {
-		lop = LOCK_EX | LOCK_NB;
+		lock.l_type = F_WRLCK;
 	} else {
-		lop = LOCK_SH | LOCK_NB;
+		lock.l_type = F_RDLCK;
 	}
 
 	for (i = 0; i < LOCK_RETRIES; i++) {
-		if (flock(fh, lop) == 0)
+		if (fcntl(fh, F_SETLK, &lock) != -1)
 			break;
 		sleep(1);
 	}
diff -ur Argyll_V1.3.2.old/jcnf/jcnf.h Argyll_V1.3.2/jcnf/jcnf.h
--- Argyll_V1.3.2.old/jcnf/jcnf.h	2010-11-03 05:19:31.000000000 -0400
+++ Argyll_V1.3.2/jcnf/jcnf.h	2011-01-03 17:42:04.787044048 -0500
@@ -43,6 +43,10 @@
 
  */
 
+#ifdef __sun
+#include <sys/time.h>
+#endif
+
 /* jcnf error codes */
 typedef enum {
 	jc_ok		    = 0,		/* No error */
diff -ur Argyll_V1.3.2.old/numlib/numsup.c Argyll_V1.3.2/numlib/numsup.c
--- Argyll_V1.3.2.old/numlib/numsup.c	2010-11-03 05:18:05.000000000 -0400
+++ Argyll_V1.3.2/numlib/numsup.c	2011-01-03 17:42:21.878101167 -0500
@@ -22,6 +22,7 @@
 #ifdef UNIX
 #include <unistd.h>
 #include <sys/param.h>
+#include <limits.h>
 #endif
 
 #include "numsup.h"
diff -ur Argyll_V1.3.2.old/plot/Jamfile Argyll_V1.3.2/plot/Jamfile
--- Argyll_V1.3.2.old/plot/Jamfile	2010-11-03 05:18:57.000000000 -0400
+++ Argyll_V1.3.2/plot/Jamfile	2011-01-03 17:42:49.753243731 -0500
@@ -5,6 +5,8 @@
 PREF_CCFLAGS	= $(CCDEBUGFLAG) ;		# Debugging flags
 PREF_LINKFLAGS	= $(LINKDEBUGFLAG) ;
 
+LINKLIBS = ../numlib/libnum ;
+
 # PLOT library
 Library libplot : plot.c : : : ../h ../numlib ../spectro ;
 if $(UNIX) && $(OS) != MACOSX {
diff -ur Argyll_V1.3.2.old/spectro/hidio.c Argyll_V1.3.2/spectro/hidio.c
--- Argyll_V1.3.2.old/spectro/hidio.c	2010-11-03 05:19:14.000000000 -0400
+++ Argyll_V1.3.2/spectro/hidio.c	2011-01-03 17:48:42.497006429 -0500
@@ -58,6 +58,8 @@
 #if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <sys/types.h> 
 #include <usbhid.h> 
+#elif defined(__sun)
+#include <sys/usb/clients/hid/hid.h>
 #else	/* assume Linux */ 
 #include <asm/types.h>
 #include <linux/hiddev.h>
diff -ur Argyll_V1.3.2.old/spectro/Jamfile Argyll_V1.3.2/spectro/Jamfile
--- Argyll_V1.3.2.old/spectro/Jamfile	2010-11-03 05:19:11.000000000 -0400
+++ Argyll_V1.3.2/spectro/Jamfile	2011-01-03 17:46:40.355598830 -0500
@@ -27,21 +27,6 @@
 	IOFILE = ntio.c ;
 }
 if $(UNIX) {
-	if $(USE_LIBUSB1) = true {
-		LIBUSBDIR = ../libusb1 ;
-		LIBUSBHDRS = ../libusb1 ;
-		if $(LIBUSB_IS_DLL) = true {
-			LIBUSB = $(LIBUSB1NAME)$(SUFIMPLIB) ;
-			LIBUSBSH = $(LIBUSB1NAME)$(SUFSHLIB) ;
-		} else {
-			LIBUSB = $(LIBUSB1NAME)$(SUFLIB) ;
-		}
-		DEFINES += USE_LIBUSB1 ;
-	} else {
-		LIBUSBDIR = ../libusb ;
-		LIBUSBHDRS = ../libusb ;
-		LIBUSB = libusb ;
-	}
 	IOFILE = unixio.c ;
 	CONVFILE = pollem.c ;
 }
@@ -67,7 +52,7 @@
 }
 
 HDRS = ../h ../numlib ../icc ../cgats ../rspl ../xicc ../gamut ../spectro
-       ../plot $(LIBUSBHDRS) $(CMMHDRS) ;
+       ../plot $(CMMHDRS) ;
 
 # Instrument access library library
 Library libinst : inst.c insttypes.c dtp20.c dtp22.c dtp41.c dtp51.c dtp92.c i1disp.c i1pro.c i1pro_imp.c munki.c munki_imp.c ss.c ss_imp.c hcfr.c spyd2.c huey.c $(IOFILE) usbio.c hidio.c ;
@@ -98,8 +83,6 @@
 	File $(LIBUSBSH) : $(LIBUSBDIR)/$(LIBUSBSH) ;
 	# executable needs .so/.dll in same directory
 	NDepends $(Executables) : $(LIBUSBSH) ;
-} else {
-	LINKLIBS += $(LIBUSBDIR)/$(LIBUSB) ;
 }
 
 # General target reader program
diff -ur Argyll_V1.3.2.old/spectro/usbio.c Argyll_V1.3.2/spectro/usbio.c
--- Argyll_V1.3.2.old/spectro/usbio.c	2010-11-03 05:19:14.000000000 -0400
+++ Argyll_V1.3.2/spectro/usbio.c	2011-01-03 17:49:24.831257651 -0500
@@ -178,10 +178,6 @@
 #ifdef ENABLE_USB
 	struct usb_bus *bus;
 
-	/* Check that we've got an up to date version of libusb */
-	if (usb_argyll_patched() < 2)
-		error("usblib isn't up to date to work with this version of Argyll");
-
 	if (p->debug > 8)
 		usb_set_debug(p->debug);
 
diff -ur Argyll_V1.3.2.old/ucmm/ucmm.c Argyll_V1.3.2/ucmm/ucmm.c
--- Argyll_V1.3.2.old/ucmm/ucmm.c	2010-11-03 05:19:30.000000000 -0400
+++ Argyll_V1.3.2/ucmm/ucmm.c	2011-01-03 17:49:52.389631267 -0500
@@ -763,7 +763,7 @@
 				}
 				if ((pp = jc_get_nth_elem(key, 3)) == NULL)
 					continue;
-				if (strcmp(pp,"ICC_PROFILE") != 0i
+				if (strcmp(pp,"ICC_PROFILE") != 0
 				 || type != jc_string
 				 || strcmp(data, data_name) != 0) {
 					free(pp);
