--- include/curl/curlbuild.h.in.orig	2008-10-28 17:00:57.788131395 -0400
+++ include/curl/curlbuild.h.in	2008-10-28 17:02:18.907768951 -0400
@@ -58,11 +58,6 @@
 /*  DEFINITION OF THESE SYMBOLS SHALL NOT TAKE PLACE ANYWHERE ELSE  */
 /* ================================================================ */
 
-#ifdef CURL_SIZEOF_LONG
-#  error "CURL_SIZEOF_LONG shall not be defined except in curlbuild.h"
-   Error Compilation_aborted_CURL_SIZEOF_LONG_already_defined
-#endif
-
 #ifdef CURL_TYPEOF_CURL_OFF_T
 #  error "CURL_TYPEOF_CURL_OFF_T shall not be defined except in curlbuild.h"
    Error Compilation_aborted_CURL_TYPEOF_CURL_OFF_T_already_defined
@@ -123,9 +118,6 @@
 #  include <inttypes.h>
 #endif
 
-/* The size of `long', as computed by sizeof. */
-#undef CURL_SIZEOF_LONG
-
 /* Signed integral data type used for curl_off_t. */
 #undef CURL_TYPEOF_CURL_OFF_T
 
@@ -150,4 +142,10 @@
 /* unsigned curl_off_t constant suffix. */
 #undef CURL_SUFFIX_CURL_OFF_TU
 
+#if defined(__arch64__) || defined(__x86_64) || defined(__sparcv9)
+#define CURL_SIZEOF_LONG 8
+#else
+#define CURL_SIZEOF_LONG 4
+#endif 
+
 #endif /* __CURL_CURLBUILD_H */
--- configure.ac.orig	2008-10-28 17:10:55.883946335 -0400
+++ configure.ac	2008-10-28 17:11:11.818282990 -0400
@@ -1998,7 +1998,6 @@
 AC_CHECK_SIZEOF(size_t)
 AC_CHECK_SIZEOF(int)
 AC_CHECK_SIZEOF(long)
-CURL_CONFIGURE_LONG
 AC_CHECK_SIZEOF(time_t)
 AC_CHECK_SIZEOF(off_t)
 
