diff -ur exiv2-0.21.old/config/config.mk.in exiv2-0.21/config/config.mk.in
--- exiv2-0.21.old/config/config.mk.in	2010-04-03 03:53:30.000000000 -0400
+++ exiv2-0.21/config/config.mk.in	2011-01-12 17:35:26.711674551 -0500
@@ -153,7 +153,7 @@
 COMPILE.c = $(CC) $(CFLAGS) $(CPPFLAGS) -c
 # LINK.cc does not need $(LIBS), libtool's dark magic takes care of that
 # when linking a binary with a libtool library.
-LINK.cc = $(CXX) $(LDFLAGS)
+LINK.cc = $(CXX) $(CXXFLAGS) $(LDFLAGS)
 
 # **********************************************************************
 # Installation programs
diff -ur exiv2-0.21.old/src/basicio.cpp exiv2-0.21/src/basicio.cpp
--- exiv2-0.21.old/src/basicio.cpp	2010-09-10 10:02:09.000000000 -0400
+++ exiv2-0.21/src/basicio.cpp	2011-01-12 17:35:26.706520850 -0500
@@ -269,7 +269,7 @@
         int rc = 0;
         if (p_->pMappedArea_ != 0) {
 #if defined EXV_HAVE_MMAP && defined EXV_HAVE_MUNMAP
-            if (::munmap(p_->pMappedArea_, p_->mappedLength_) != 0) {
+            if (::munmap((char *) p_->pMappedArea_, p_->mappedLength_) != 0) {
                 rc = 1;
             }
 #elif defined WIN32 && !defined __CYGWIN__
diff -ur exiv2-0.21.old/src/convert.cpp exiv2-0.21/src/convert.cpp
--- exiv2-0.21.old/src/convert.cpp	2010-09-10 10:02:09.000000000 -0400
+++ exiv2-0.21/src/convert.cpp	2011-01-12 17:35:26.694955907 -0500
@@ -578,7 +578,8 @@
             double dhour = pos->toFloat(0);
             double dmin = pos->toFloat(1);
             // Hack: Need Value::toDouble
-            URational r = pos->toRational(2);
+            Rational tempr = pos->toRational(2);
+            URational r(tempr.first, tempr.second);
             double dsec = static_cast<double>(r.first)/r.second;
 
             if (!pos->value().ok()) {
diff -ur exiv2-0.21.old/src/crwimage.cpp exiv2-0.21/src/crwimage.cpp
--- exiv2-0.21.old/src/crwimage.cpp	2010-08-30 15:33:07.000000000 -0400
+++ exiv2-0.21/src/crwimage.cpp	2011-01-12 17:35:26.706138604 -0500
@@ -1035,7 +1035,8 @@
         if (ifdId == canonSiId) {
             // Exif.Photo.FNumber
             float f = fnumber(canonEv(aperture));
-            URational ur = floatToRationalCast(f);
+            Rational tempur = floatToRationalCast(f);
+            URational ur(tempur.first, tempur.second);
             URationalValue fn;
             fn.value_.push_back(ur);
             image.exifData().add(ExifKey("Exif.Photo.FNumber"), &fn);
diff -ur exiv2-0.21.old/src/error.cpp exiv2-0.21/src/error.cpp
--- exiv2-0.21.old/src/error.cpp	2010-09-21 11:45:07.000000000 -0400
+++ exiv2-0.21/src/error.cpp	2011-01-12 18:02:10.351601909 -0500
@@ -26,7 +26,7 @@
  */
 // *****************************************************************************
 #include "rcsid_int.hpp"
-EXIV2_RCSID("@(#) $Id: error.cpp 2455 2011-02-13 14:39:15Z ahuggel $")
+//EXIV2_RCSID("@(#) $Id: error.cpp 2455 2011-02-13 14:39:15Z ahuggel $")
 
 // *****************************************************************************
 // included header files
diff -ur exiv2-0.21.old/src/error.hpp exiv2-0.21/src/error.hpp
--- exiv2-0.21.old/src/error.hpp	2010-09-10 10:02:09.000000000 -0400
+++ exiv2-0.21/src/error.hpp	2011-01-14 13:21:32.747473251 -0500
@@ -99,6 +99,13 @@
         }
         //@}
 
+        // Fix bug in warnings called from tiffcomposite.cpp
+        LogMsg& operator<<(std::ios_base& (*pf)(std::ios_base&))
+        {
+            pf(os_);
+            return *this;
+        }
+
         /*!
           @brief Set the log level. Only log messages with a level greater or
                  equal \em level are sent to the log message handler. Default
diff -ur exiv2-0.21.old/src/exif.cpp exiv2-0.21/src/exif.cpp
--- exiv2-0.21.old/src/exif.cpp	2010-09-21 11:45:07.000000000 -0400
+++ exiv2-0.21/src/exif.cpp	2011-01-14 13:02:23.629631433 -0500
@@ -599,14 +599,37 @@
         exifMetadata_.clear();
     }
 
+    template <class T, class S>
+    void sort(std::list<T>& list, bool (*cmp)(const S& t1, const S& t2)) {
+        bool swapped = true;
+        int j = 0;
+
+        while (swapped) {
+            swapped = false;
+            j++;
+            std::list<T>::iterator it = list.begin();
+            for (int i = 0; i < list.size() - j; i++) {
+                std::list<T>::iterator next = it;
+                next++;
+                if (cmp(*next, *it)) {
+                    list.insert(it, *next);
+                    list.erase(next);
+                    swapped = true;
+                } else {
+                    it++;
+                }
+            }
+        }
+    }
+
     void ExifData::sortByKey()
     {
-        exifMetadata_.sort(cmpMetadataByKey);
+        sort(exifMetadata_, cmpMetadataByKey);
     }
 
     void ExifData::sortByTag()
     {
-        exifMetadata_.sort(cmpMetadataByTag);
+        sort(exifMetadata_, cmpMetadataByTag);
     }
 
     ExifData::iterator ExifData::erase(ExifData::iterator beg, ExifData::iterator end)
diff -ur exiv2-0.21.old/src/exiv2.cpp exiv2-0.21/src/exiv2.cpp
--- exiv2-0.21.old/src/exiv2.cpp	2010-11-06 09:53:15.000000000 -0400
+++ exiv2-0.21/src/exiv2.cpp	2011-01-12 17:35:26.703399750 -0500
@@ -53,6 +53,10 @@
 #include <cassert>
 #include <cctype>
 
+#ifdef EXV_ENABLE_NLS
+# include <clocale>
+#endif
+
 // *****************************************************************************
 // local declarations
 namespace {
@@ -131,7 +135,7 @@
 int main(int argc, char* const argv[])
 {
 #ifdef EXV_ENABLE_NLS
-    setlocale(LC_ALL, "");
+    std::setlocale(LC_ALL, "");
     bindtextdomain(EXV_PACKAGE, EXV_LOCALEDIR);
     textdomain(EXV_PACKAGE);
 #endif
diff -ur exiv2-0.21.old/src/jpgimage.cpp exiv2-0.21/src/jpgimage.cpp
--- exiv2-0.21.old/src/jpgimage.cpp	2010-09-10 10:02:09.000000000 -0400
+++ exiv2-0.21/src/jpgimage.cpp	2011-01-12 17:35:26.707518967 -0500
@@ -782,7 +782,7 @@
             }
             else if (   skipApp1Exif == count
                      || skipApp1Xmp  == count
-                     || find(skipApp13Ps3.begin(), skipApp13Ps3.end(), count) != skipApp13Ps3.end()
+                     || std::find(skipApp13Ps3.begin(), skipApp13Ps3.end(), count) != skipApp13Ps3.end()
                      || skipCom      == count) {
                 --search;
                 io_->seek(size-bufRead, BasicIo::cur);
diff -ur exiv2-0.21.old/src/Makefile exiv2-0.21/src/Makefile
--- exiv2-0.21.old/src/Makefile	2010-11-22 10:44:15.000000000 -0500
+++ exiv2-0.21/src/Makefile	2011-01-12 17:35:26.694631229 -0500
@@ -172,7 +172,7 @@
 EXECUTABLE = $(BINSRC:.cpp=$(EXEEXT))
 
 EXIV2OBJ = $(EXIV2MAIN:.cpp=.o) $(EXIV2SRC:.cpp=.o)
-EXIV2COBJ = $(EXIVCSRC:.c=.o)
+EXIV2COBJ = $(EXIVCSRC:.c=.lo)
 EXIV2BIN = $(EXIV2MAIN:.cpp=)
 EXIV2EXE = $(EXIV2MAIN:.cpp=$(EXEEXT))
 
diff -ur exiv2-0.21.old/xmpsdk/src/XMPCore_Impl.hpp exiv2-0.21/xmpsdk/src/XMPCore_Impl.hpp
--- exiv2-0.21.old/xmpsdk/src/XMPCore_Impl.hpp	2010-06-13 10:21:05.000000000 -0400
+++ exiv2-0.21/xmpsdk/src/XMPCore_Impl.hpp	2011-01-12 17:35:26.760612331 -0500
@@ -52,7 +52,7 @@
 typedef XMP_VarString::iterator			XMP_VarStringPos;
 typedef XMP_VarString::const_iterator	XMP_cVarStringPos;
 
-typedef std::pair < XMP_VarString, XMP_VarString >	XMP_StringPair;
+typedef std::pair < const XMP_VarString, XMP_VarString >	XMP_StringPair;
 
 typedef std::map < XMP_VarString, XMP_VarString > XMP_StringMap;
 typedef XMP_StringMap::iterator			XMP_StringMapPos;
