--- libavcodec/mlib/dsputil_mlib.c.old	Fri Feb 12 14:42:38 2010
+++ libavcodec/mlib/dsputil_mlib.c	Fri Feb 12 14:42:44 2010
@@ -416,7 +416,10 @@
 
 static void ff_fdct_mlib(DCTELEM *data)
 {
+    static const mlib_s16 c = 8;
+
     mlib_VideoDCT8x8_S16_S16 (data, data);
+    mlib_VectorMulS_S16_Sat (data, &c, 64); /* scaling data by 8 */
 }
 
 void dsputil_init_mlib(DSPContext* c, AVCodecContext *avctx)
--- libswscale/mlib/yuv2rgb_mlib.c.old	Fri Feb 12 14:41:58 2010
+++ libswscale/mlib/yuv2rgb_mlib.c	Fri Feb 12 14:41:30 2010
@@ -29,6 +29,7 @@
 #include <assert.h>
 
 #include "libswscale/swscale.h"
+#include "libswscale/swscale_internal.h"
 
 static int mlib_YUV2ARGB420_32(SwsContext *c, uint8_t* src[], int srcStride[], int srcSliceY,
                                int srcSliceH, uint8_t* dst[], int dstStride[])
