--- configure.old	Fri Feb 12 15:59:06 2010
+++ configure	Fri Feb 12 15:59:39 2010
@@ -2585,7 +2585,7 @@
     check_struct SDL.h SDL_VideoInfo current_w $sdl_cflags && enable sdl_video_size
 fi
 
-texi2html -version > /dev/null 2>&1 && enable texi2html || disable texi2html
+(texi2html -version > /dev/null 2>&1 || texi2html /dev/null > /dev/null 2>&1) && enable texi2html || disable texi2html
 
 if enabled network; then
     check_type "sys/types.h sys/socket.h" socklen_t
