diff -Nur openafs-1.4.8.orig/configure openafs-1.4.8/configure
--- openafs-1.4.8.orig/configure	2009-01-31 23:19:12.292202651 -0500
+++ openafs-1.4.8/configure	2009-02-01 14:32:25.114272111 -0500
@@ -6460,6 +6460,9 @@
 		i386-pc-solaris2.11)
 			AFS_SYSNAME="sunx86_511"
 			;;
+		amd64-pc-solaris2.11)
+			AFS_SYSNAME="sunamd64_511"
+			;;
 		alpha*-dec-osf4.0*)
 			AFS_SYSNAME="alpha_dux40"
 			;;
@@ -24026,11 +24029,11 @@
 		;;
 
 	sun4x_511)
-		CC="/opt/SUNWspro/bin/cc"
-		CCOBJ="/opt/SUNWspro/bin/cc"
+		CC="/opt/SunStudioExpress/bin/cc"
+		CCOBJ="/opt/SunStudioExpress/bin/cc"
 		LEX="lex"
 		LD="/usr/ccs/bin/ld"
-		MT_CC="/opt/SUNWspro/bin/cc"
+		MT_CC="/opt/SunStudioExpress/bin/cc"
 		MT_CFLAGS='-mt -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-lpthread -lsocket"
 		PAM_CFLAGS="-KPIC"
@@ -24132,11 +24135,11 @@
 		;;
 
 	sunx86_511)
-		CC="/opt/SUNWspro/bin/cc"
-		CCOBJ="/opt/SUNWspro/bin/cc"
+		CC="/opt/SunStudioExpress/bin/cc"
+		CCOBJ="/opt/SunStudioExpress/bin/cc"
 		LEX="lex"
 		LD="/usr/ccs/bin/ld"
-		MT_CC="/opt/SUNWspro/bin/cc"
+		MT_CC="/opt/SunStudioExpress/bin/cc"
 		MT_CFLAGS='-mt -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-lpthread -lsocket"
 		PAM_CFLAGS="-KPIC"
@@ -24151,6 +24154,29 @@
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
 		SHLIB_LINKER="${CC} -G -dy -Bsymbolic -z text"
 		;;
+
+	sunamd64_511)
+		CC="/opt/SunStudioExpress/bin/cc -m64"
+		CCOBJ="/opt/SunStudioExpress/bin/cc -m64"
+		LEX="lex"
+		LD="/usr/ccs/bin/ld"
+		AS="/usr/ccs/bin/as -m64"
+		MT_CC="/opt/SunStudioExpress/bin/cc -m64"
+		MT_CFLAGS='-mt -DAFS_PTHREAD_ENV ${XCFLAGS}'
+		MT_LIBS="-lpthread -lsocket"
+		PAM_CFLAGS="-KPIC"
+		PAM_LIBS="-lc -lpam -lsocket -lnsl -lm"
+		SHLIB_CFLAGS="-KPIC"
+		SHLIB_LDFLAGS="-G -Bsymbolic"
+		TXLIBS="-lcurses"
+		XCFLAGS64='${XCFLAGS} -m64'
+		XCFLAGS="-dy -Bdynamic -Kpic"
+		XLIBELFA="-lelf"
+		XLIBKVM="-lkvm"
+		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
+		SHLIB_LINKER="${CC} -G -dy -Bsymbolic -z text"
+		;;
+	
 esac
 
 #
@@ -24250,7 +24276,7 @@
 
 	;;
 
-	sun4x_*|sunx86_*)
+	sun4x_*|sunx86_*|sunamd64_*)
 		FS_CONV_SOL26="fs_conv_sol26"
 		install_FS_CONV_SOL26='$(DESTDIR)${afssrvsbindir}/fs_conv_sol26'
 		dest_FS_CONV_SOL26='$(DEST)/root.server/usr/afs/bin/fs_conv_sol26'
diff -Nur openafs-1.4.8.orig/configure-libafs openafs-1.4.8/configure-libafs
--- openafs-1.4.8.orig/configure-libafs	2009-01-31 23:19:12.291356658 -0500
+++ openafs-1.4.8/configure-libafs	2009-02-01 14:36:48.750415836 -0500
@@ -6413,6 +6413,9 @@
 		i386-pc-solaris2.11)
 			AFS_SYSNAME="sunx86_511"
 			;;
+		amd64-pc-solaris2.11)
+			AFS_SYSNAME="sunamd64_511"
+			;;
 		alpha*-dec-osf4.0*)
 			AFS_SYSNAME="alpha_dux40"
 			;;
@@ -24084,11 +24087,11 @@
 		;;
 
 	sunx86_511)
-		CC="/opt/SUNWspro/bin/cc"
-		CCOBJ="/opt/SUNWspro/bin/cc"
+		CC="/opt/SunStudioExpress/bin/cc"
+		CCOBJ="/opt/SunStudioExpress/bin/cc"
 		LEX="lex"
 		LD="/usr/ccs/bin/ld"
-		MT_CC="/opt/SUNWspro/bin/cc"
+		MT_CC="/opt/SunStudioExpress/bin/cc"
 		MT_CFLAGS='-mt -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-lpthread -lsocket"
 		PAM_CFLAGS="-KPIC"
@@ -24103,6 +24106,29 @@
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
 		SHLIB_LINKER="${CC} -G -dy -Bsymbolic -z text"
 		;;
+	
+	sunamd64_511)
+		CC="/opt/SunStudioExpress/bin/cc -m64"
+		CCOBJ="/opt/SunStudioExpress/bin/cc -m64"
+		LEX="lex"
+		LD="/usr/ccs/bin/ld"
+		AS="/usr/ccs/bin/as -m64"
+		MT_CC="/opt/SunStudioExpress/bin/cc -m64"
+		MT_CFLAGS='-mt -DAFS_PTHREAD_ENV ${XCFLAGS}'
+		MT_LIBS="-lpthread -lsocket"
+		PAM_CFLAGS="-KPIC"
+		PAM_LIBS="-lc -lpam -lsocket -lnsl -lm"
+		SHLIB_CFLAGS="-KPIC"
+		SHLIB_LDFLAGS="-G -Bsymbolic"
+		TXLIBS="-lcurses"
+		XCFLAGS64='${XCFLAGS} -m64'
+		XCFLAGS="-dy -Bdynamic -Kpic"
+		XLIBELFA="-lelf"
+		XLIBKVM="-lkvm"
+		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
+		SHLIB_LINKER="${CC} -G -dy -Bsymbolic -z text"
+		;;
+
 esac
 
 #
@@ -24202,7 +24228,7 @@
 
 	;;
 
-	sun4x_*|sunx86_*)
+	sun4x_*|sunx86_*|sunamd64_*)
 		FS_CONV_SOL26="fs_conv_sol26"
 		install_FS_CONV_SOL26='$(DESTDIR)${afssrvsbindir}/fs_conv_sol26'
 		dest_FS_CONV_SOL26='$(DEST)/root.server/usr/afs/bin/fs_conv_sol26'
diff -Nur openafs-1.4.8.orig/Makefile.in openafs-1.4.8/Makefile.in
--- openafs-1.4.8.orig/Makefile.in	2009-01-31 23:19:19.682902542 -0500
+++ openafs-1.4.8/Makefile.in	2009-02-02 17:05:14.932344089 -0500
@@ -259,7 +259,7 @@
 ${TOP_LIBDIR}/libtermlib.a: 
 	$(RM) -f ${TOP_LIBDIR}/libtermlib.a
 	case ${SYS_NAME} in \
-	rs_aix* | sun4x_* | sunx86_* | sgi_6? | *linux* | *_darwin* | *fbsd* | ia64_hpux*) \
+	rs_aix* | sun4x_* | sunx86_* | sunamd64_* | sgi_6? | *linux* | *_darwin* | *fbsd* | ia64_hpux*) \
 		ln -s libnull.a ${TOP_LIBDIR}/libtermlib.a ;; \
 	*) \
 		ln -s /usr/lib/libtermlib.a ${TOP_LIBDIR}/libtermlib.a ;; \
@@ -464,7 +464,7 @@
 
 tbutc: cmd comerr bubasics butm budb bucoord cmd butc libadmin
 	case ${SYS_NAME} in \
-	alpha_dux*|sgi_*|sun4x_*|sunx86_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*) \
+	alpha_dux*|sgi_*|sun4x_*|sunx86_*|sunamd64_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*) \
 		${COMPILE_PART1} tbutc ${COMPILE_PART2} ;; \
 	*) \
 		echo Not building MT butc for ${SYS_NAME} ;; \
@@ -487,7 +487,7 @@
 # pthread based user space RX library
 libafsrpc: rx rxkad des
 	case ${SYS_NAME} in \
-	alpha_dux*|sgi_*|sun4x_*|sunx86_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*|*[of]bsd*|*nbsd[234]*) \
+	alpha_dux*|sgi_*|sun4x_*|sunx86_*|sunamd64_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*|*[of]bsd*|*nbsd[234]*) \
 	${COMPILE_PART1} libafsrpc ${COMPILE_PART2} ;; \
 	*_darwin_[1-6][0-9]) \
 		echo Not building MT libafsrpc for ${SYS_NAME} ;; \
@@ -499,7 +499,7 @@
 
 libafsauthent: ubik auth kauth libafsrpc
 	case ${SYS_NAME} in \
-	alpha_dux*|sgi_*|sun4x_*|sunx86_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*|*[of]bsd*|*nbsd[234]*) \
+	alpha_dux*|sgi_*|sun4x_*|sunx86_*|sunamd64_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*|*[of]bsd*|*nbsd[234]*) \
 	${COMPILE_PART1} libafsauthent ${COMPILE_PART2} ;; \
 	*_darwin_[1-6][0-9]) \
 		echo Not building MT libafsauthent for ${SYS_NAME} ;; \
@@ -514,7 +514,7 @@
 	case ${SYS_NAME} in \
 	amd64_linux24) \
 		echo Skipping shlibafsrpc for amd64_linux24 ;; \
-	alpha_dux*|sgi_*|sun4x_*|sunx86_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*) \
+	alpha_dux*|sgi_*|sun4x_*|sunx86_*|sunamd64_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*) \
 	${COMPILE_PART1} shlibafsrpc ${COMPILE_PART2} ;; \
 	*) \
 		echo Not building shared libafsrpc for ${SYS_NAME} ;; \
@@ -524,7 +524,7 @@
 	case ${SYS_NAME} in \
 	amd64_linux24) \
 		echo Skipping shlibafsauthent for amd64_linux24 ;; \
-	alpha_dux*|sgi_*|sun4x_*|sunx86_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*) \
+	alpha_dux*|sgi_*|sun4x_*|sunx86_*|sunamd64_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*) \
 	${COMPILE_PART1} shlibafsauthent ${COMPILE_PART2} ;; \
 	*) \
 		echo Not building shared libafsauthent for ${SYS_NAME} ;; \
@@ -544,7 +544,7 @@
 
 libadmin: libafsauthent bozo
 	case ${SYS_NAME} in \
-	alpha_dux*|sgi_*|sun4x_*|sunx86_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*) \
+	alpha_dux*|sgi_*|sun4x_*|sunx86_*|sunamd64_*|rs_aix*|*linux*|hp_ux11*|ia64_hpux*) \
 			$(MAKE) libadmin_real ;; \
 	*_darwin_[1-6][0-9]) \
 		echo Not building MT libadmin for ${SYS_NAME} ;; \
diff -Nur openafs-1.4.8.orig/src/butc/lwps.c openafs-1.4.8/src/butc/lwps.c
--- openafs-1.4.8.orig/src/butc/lwps.c	2009-01-31 23:19:12.890066022 -0500
+++ openafs-1.4.8/src/butc/lwps.c	2009-01-31 23:20:50.863828821 -0500
@@ -347,7 +347,7 @@
     if (stdin->_IO_read_ptr < stdin->_IO_read_end)
 	return 1;
 #else
-    if (stdin->_cnt > 0)
+    if (getc_unlocked(stdin) > 0)
 	return 1;
 #endif
     FD_ZERO(&rdfds);
diff -Nur openafs-1.4.8.orig/src/config/afs_sysnames.h openafs-1.4.8/src/config/afs_sysnames.h
--- openafs-1.4.8.orig/src/config/afs_sysnames.h	2009-01-31 23:19:13.111585786 -0500
+++ openafs-1.4.8/src/config/afs_sysnames.h	2009-02-01 00:20:40.215175540 -0500
@@ -110,6 +110,8 @@
 #define SYS_NAME_ID_sunx86_510		 953
 #define SYS_NAME_ID_sunx86_511		 954
 
+#define SYS_NAME_ID_sunamd64_511	 970
+
 #define SYS_NAME_ID_vax_ul4		1003
 #define SYS_NAME_ID_vax_ul40		1004
 #define SYS_NAME_ID_vax_ul43		1005
diff -Nur openafs-1.4.8.orig/src/config/param.sunamd64_511.h openafs-1.4.8/src/config/param.sunamd64_511.h
--- openafs-1.4.8.orig/src/config/param.sunamd64_511.h	1969-12-31 19:00:00.000000000 -0500
+++ openafs-1.4.8/src/config/param.sunamd64_511.h	2009-01-31 23:37:11.788435390 -0500
@@ -0,0 +1,190 @@
+#ifndef UKERNEL
+/* This section for kernel libafs compiles only */
+
+/*
+ * Copyright 2000, International Business Machines Corporation and others.
+ * All Rights Reserved.
+ * 
+ * This software has been released under the terms of the IBM Public
+ * License.  For details, see the LICENSE file in the top-level source
+ * directory or online at http://www.openafs.org/dl/license10.html
+ */
+
+#ifndef	AFS_PARAM_H
+#define	AFS_PARAM_H
+
+#define AFS_VFS_ENV		1
+/* Used only in vfsck code; is it needed any more???? */
+
+#define AFS_VFSINCL_ENV		1	/* NOBODY uses this.... */
+#define AFS_GREEDY43_ENV	1	/* Used only in rx/rx_user.c */
+#define AFS_ENV			1
+#define AFS_SUN_ENV		1
+#define AFS_SUN5_ENV		1
+#define	AFS_SUN52_ENV		1
+#define	AFS_SUN53_ENV		1
+#define	AFS_SUN54_ENV		1
+#define	AFS_SUN55_ENV		1
+#define	AFS_SUN56_ENV		1
+#define AFS_SUN57_ENV		1
+#define AFS_SUN58_ENV		1
+#define AFS_SUN59_ENV		1
+#define AFS_SUN510_ENV		1
+#define AFS_SUN511_ENV		1
+#define AFS_BOZONLOCK_ENV       1
+#define AFS_X86_ENV		1
+
+#define AFS_64BIT_ENV		1	/* Defines afs_int32 as int, not long. */
+#define AFS_64BIT_CLIENT	1	
+
+#define AFS_HAVE_FLOCK_SYSID    1
+
+#include <afs/afs_sysnames.h>
+
+#define AFS_GLOBAL_SUNLOCK	1	/* For global locking */
+#define RXK_LISTENER_ENV	1
+#define AFS_GCPAGS		1	/* if nonzero, garbage collect PAGs */
+
+#ifdef AFS_NAMEI_ENV
+#define AFS_64BIT_IOPS_ENV	1	/* needed for NAMEI... */
+#else /* AFS_NAMEI_ENV */
+#define	AFS_3DISPARES		1	/* Utilize the 3 available disk inode 'spares' */
+#endif /* AFS_NAMEI_ENV */
+
+#define	AFS_SYSCALL		65
+
+/* File system entry (used if mount.h doesn't define MOUNT_AFS */
+#define AFS_MOUNT_AFS		"afs"
+
+/* Machine / Operating system information */
+#define sys_sunamd64_511	1
+#define SYS_NAME		"sunamd64_511"
+#define SYS_NAME_ID		SYS_NAME_ID_sunamd64_511
+#define AFSLITTLE_ENDIAN	1
+#define AFS_HAVE_FFS		1	/* Use system's ffs. */
+#define AFS_HAVE_VXFS		1	/* Support cache on Veritas vxfs file system */
+#define AFS_HAVE_STATVFS	1	/* System supports statvfs */
+#define AFS_VM_RDWR_ENV		1	/* read/write implemented via VM */
+#define AFS_USE_GETTIMEOFDAY	1	/* use gettimeofday to implement rx clock */
+
+#define NEARINODE_HINT		1	/* hint to ufs module to scatter inodes on disk */
+#define nearInodeHash(volid, hval) {                                 \
+                unsigned char*  ts = (unsigned char*)&(volid)+sizeof(volid)-1;\
+                for ( (hval)=0; ts >= (unsigned char*)&(volid); ts--){\
+                    (hval) *= 173;                      \
+                    (hval) += *ts;                      \
+                }                                       \
+                }
+
+/* Extra kernel definitions (from kdefs file) */
+#ifdef KERNEL
+/* sun definitions here */
+#define	AFS_UIOFMODE		1	/* Only in afs/afs_vnodeops.c (afs_ustrategy) */
+#define	AFS_SYSVLOCK		1	/* sys v locking supported */
+/*#define	AFS_USEBUFFERS	1*/
+#define	afsio_iov		uio_iov
+#define	afsio_iovcnt		uio_iovcnt
+#define	afsio_offset		uio_loffset
+#define	afsio_seg		uio_segflg
+#define	afsio_fmode		uio_fmode
+#define	afsio_resid		uio_resid
+#define	AFS_UIOSYS		UIO_SYSSPACE
+#define	AFS_UIOUSER		UIO_USERSPACE
+#define	AFS_CLBYTES		MCLBYTES
+#define	AFS_MINCHANGE		2
+#define	osi_GetTime(x)		uniqtime32(x)
+
+/**
+  * These defines are for the 64 bit Solaris 7 port
+  * AFS_SYSCALL32 is used to protect the ILP32 syscall interface
+  * AFS_64BIT_ENV is for use of 64 bit inode numbers
+  */
+#define __amd64			1 
+#if defined(__amd64)
+#define	AFS_SUN57_64BIT_ENV	1
+#define AFS_64BIT_INO   	1
+#endif
+
+#define	AFS_KALLOC(n)		kmem_alloc(n, KM_SLEEP)
+#define AFS_KALLOC_NOSLEEP(n)   kmem_alloc(n, KM_NOSLEEP)
+#define	AFS_KFREE		kmem_free
+#define	VATTR_NULL		vattr_null
+#endif /* KERNEL */
+#define	AFS_DIRENT
+#ifndef CMSERVERPREF
+#define CMSERVERPREF
+#endif
+#define	ROOTINO			UFSROOTINO
+
+#endif /* AFS_PARAM_H */
+
+#else /* !defined(UKERNEL) */
+
+/* This section for user space compiles only */
+
+#ifndef	AFS_PARAM_H
+#define	AFS_PARAM_H
+
+#define AFS_VFS_ENV		1
+/* Used only in vfsck code; is it needed any more???? */
+#define RXK_LISTENER_ENV	1
+#define AFS_USERSPACE_IP_ADDR	1
+#define AFS_GCPAGS		0	/* if nonzero, garbage collect PAGs */
+
+#define UKERNEL			1	/* user space kernel */
+#define AFS_GREEDY43_ENV	1	/* Used only in rx/rx_user.c */
+#define AFS_ENV			1
+#define AFS_USR_SUN5_ENV	1
+#define AFS_USR_SUN6_ENV	1
+#define AFS_USR_SUN7_ENV        1
+#define AFS_USR_SUN8_ENV        1
+#define AFS_USR_SUN9_ENV        1
+
+#define AFS_64BIT_ENV           1
+
+#include <afs/afs_sysnames.h>
+
+#if 0
+#define AFS_GLOBAL_SUNLOCK	1	/* For global locking */
+#endif
+
+#define	AFS_3DISPARES		1	/* Utilize the 3 available disk inode 'spares' */
+#define	AFS_SYSCALL		65
+
+/* File system entry (used if mount.h doesn't define MOUNT_AFS */
+#define AFS_MOUNT_AFS		1
+
+/* Machine / Operating system information */
+#define sys_sunamd64_511        1
+#define SYS_NAME                "sunamd64_511"
+#define SYS_NAME_ID             SYS_NAME_ID_sunamd64_511
+#define AFSLITTLE_ENDIAN	1
+#define AFS_HAVE_FFS            1	/* Use system's ffs. */
+#define AFS_HAVE_STATVFS	0	/* System doesn't support statvfs */
+
+/* Extra kernel definitions (from kdefs file) */
+#ifdef KERNEL
+#define	AFS_UIOFMODE		1	/* Only in afs/afs_vnodeops.c (afs_ustrategy) */
+#define	AFS_SYSVLOCK		1	/* sys v locking supported */
+/*#define	AFS_USEBUFFERS	1*/
+#define	afsio_iov		uio_iov
+#define	afsio_iovcnt		uio_iovcnt
+#define	afsio_offset		uio_offset
+#define	afsio_seg		uio_segflg
+#define	afsio_fmode		uio_fmode
+#define	afsio_resid		uio_resid
+#define	AFS_UIOSYS		1
+#define	AFS_UIOUSER		UIO_USERSPACE
+#define	AFS_CLBYTES		MCLBYTES
+#define	AFS_MINCHANGE		2
+#define	VATTR_NULL		usr_vattr_null
+#endif /* KERNEL */
+#define	AFS_DIRENT
+#ifndef CMSERVERPREF
+#define CMSERVERPREF
+#endif
+#define	ROOTINO			UFSROOTINO
+
+#endif /* AFS_PARAM_H */
+
+#endif /* !defined(UKERNEL) */
diff -Nur openafs-1.4.8.orig/src/libafs/MakefileProto.SOLARIS.in openafs-1.4.8/src/libafs/MakefileProto.SOLARIS.in
--- openafs-1.4.8.orig/src/libafs/MakefileProto.SOLARIS.in	2009-01-31 23:19:13.854831516 -0500
+++ openafs-1.4.8/src/libafs/MakefileProto.SOLARIS.in	2009-02-01 00:34:33.930066998 -0500
@@ -40,24 +40,28 @@
 KDEFS_32 = 
 KDEFS_64 = -xarch=amd64 -xmodel=kernel
 
+<sunamd64_511>
+KDEFS_32 = 
+KDEFS_64 = -xarch=amd64 -xmodel=kernel
+
 <all>
 CFLAGS=-I. -I.. -I${TOP_OBJDIR}/src/config ${FSINCLUDES} $(DEFINES) $(KDEFS) $(KOPTS) ${DBUG}
 
 # Name of directory to hold object files and libraries.
-<all -sun4x_57 -sun4x_58 -sun4x_59 -sun4x_510 -sun4x_511 -sunx86_57 -sunx86_58 -sunx86_59 -sunx86_510 -sunx86_511>
+<all -sun4x_57 -sun4x_58 -sun4x_59 -sun4x_510 -sun4x_511 -sunx86_57 -sunx86_58 -sunx86_59 -sunx86_510 -sunx86_511 -sunamd64_511>
 KOBJ = MODLOAD
 <sun4x_57 sun4x_58 sun4x_59 sunx86_57 sunx86_58 sunx86_59 sunx86_510 sunx86_511>
 KOBJ = MODLOAD32 MODLOAD64
-<sun4x_510 sun4x_511>
+<sun4x_510 sun4x_511 sunamd64_511>
 KOBJ = MODLOAD64
 
 # This tells Makefile.common to use it's single directory build target.
-<all -sun4x_57 -sun4x_58 -sun4x_59 -sun4x_510 -sun4x_511 -sunx86_57 -sunx86_58 -sunx86_59 -sunx86_510 -sunx86_511>
+<all -sun4x_57 -sun4x_58 -sun4x_59 -sun4x_510 -sun4x_511 -sunx86_57 -sunx86_58 -sunx86_59 -sunx86_510 -sunx86_511 -sunamd64_511>
 COMPDIRS = single_compdir
 INSTDIRS = single_instdir
 DESTDIRS = single_destdir
 
-<sun4x_57 sun4x_58 sun4x_59 sun4x_510 sun4x_511 sunx86_57 sunx86_58 sunx86_59 sunx86_510 sunx86_511>
+<sun4x_57 sun4x_58 sun4x_59 sun4x_510 sun4x_511 sunx86_57 sunx86_58 sunx86_59 sunx86_510 sunx86_511 sunamd64_511>
 COMPDIRS = solaris_compdirs
 INSTDIRS = solaris_instdirs
 DESTDIRS = solaris_destdirs
@@ -89,7 +93,7 @@
 ## This is the target for a Solaris 7. Here we build both the 32 bit and
 ## the 64 bit libafs in MODLOAD32 and MODLOAD64 directories respectively
 
-<sun4x_57 sun4x_58 sun4x_59 sun4x_510 sun4x_511 sunx86_57 sunx86_58 sunx86_59 sunx86_510 sunx86_511>
+<sun4x_57 sun4x_58 sun4x_59 sun4x_510 sun4x_511 sunx86_57 sunx86_58 sunx86_59 sunx86_510 sunx86_511 sunamd64_511>
 ${COMPDIRS} ${INSTDIRS} ${DESTDIRS}:
 	for t in ${KOBJ} ; do \
 		echo Building directory: $$t ; \
@@ -125,7 +129,7 @@
 DEST_LIBAFS = ${DEST}/root.client/usr/vice/etc/modload/${LIBAFS}
 DEST_LIBAFSNONFS = ${DEST}/root.client/usr/vice/etc/modload/${LIBAFSNONFS}
 
-<sun4x_57 sun4x_58 sun4x_59 sun4x_510 sun4x_511 sunx86_57 sunx86_58 sunx86_59 sunx86_510 sunx86_511>
+<sun4x_57 sun4x_58 sun4x_59 sun4x_510 sun4x_511 sunx86_57 sunx86_58 sunx86_59 sunx86_510 sunx86_511 sunamd64_511>
 INST_LIBAFSNONFS = ${DESTDIR}${afskerneldir}/libafs${BITS}.nonfs.o
 INST_LIBAFS = ${DESTDIR}${afskerneldir}/libafs${BITS}.o
 
diff -Nur openafs-1.4.8.orig/src/libafsrpc/Makefile.in openafs-1.4.8/src/libafsrpc/Makefile.in
--- openafs-1.4.8.orig/src/libafsrpc/Makefile.in	2009-01-31 23:19:12.929576521 -0500
+++ openafs-1.4.8/src/libafsrpc/Makefile.in	2009-02-01 14:48:17.768250602 -0500
@@ -357,6 +357,10 @@
 		$(PATH_CPP)  ${SFLAGS} ${SYS}/syscall.s > syscall.ss; \
 		as -o syscall.o syscall.ss;		\
 		$(RM) syscall.ss;;				\
+	     sunamd64_*) \
+		$(PATH_CPP)  ${SFLAGS} ${SYS}/syscall.s > syscall.ss; \
+		${AS} $(SHLIB_CFLAGS) -o syscall.o syscall.ss;		\
+		$(RM) syscall.ss;;				\
 	 sgi_* | *_darwin_* ) \
                 ${CC} ${CFLAGS} -c ${SYS}/syscall.s;;          \
 	 alpha_dux?? ) \
diff -Nur openafs-1.4.8.orig/src/lwp/Makefile.in openafs-1.4.8/src/lwp/Makefile.in
--- openafs-1.4.8.orig/src/lwp/Makefile.in	2009-01-31 23:19:12.431166454 -0500
+++ openafs-1.4.8/src/lwp/Makefile.in	2009-01-31 23:27:32.248894942 -0500
@@ -91,6 +91,10 @@
 		$(PATH_CPP) -P -D__sun -D__i386 -DIGNORE_STDS_H -I${TOP_INCDIR} -I${srcdir} ${srcdir}/process.i386.s > process.ss; \
 		${AS} -o process.o process.ss; \
 		$(RM) process.ss ;; \
+        sunamd64_*) \
+		$(CC) $(SHLIB_LDFLAGS) -P -DIGNORE_STDS_H -I${TOP_INCDIR} -I${srcdir} ${srcdir}/process.amd64.s -o process.ss; \
+		${AS} $(SHLIB_CFLAGS) -o process.o process.ss; \
+		$(RM) process.ss ;; \
 	alpha_nbsd* ) \
 		$(PATH_CPP) -DASSEMBLER -P -I${TOP_INCDIR} ${srcdir}/process.s > process.ss; \
 		${AS} -o process.o process.ss; \
diff -Nur openafs-1.4.8.orig/src/lwp/waitkey.c openafs-1.4.8/src/lwp/waitkey.c
--- openafs-1.4.8.orig/src/lwp/waitkey.c	2009-01-31 23:19:12.447407770 -0500
+++ openafs-1.4.8/src/lwp/waitkey.c	2009-01-31 23:20:33.918946760 -0500
@@ -169,7 +169,7 @@
     if (stdin->_bf._size > 0)
 	return 1;
 #else
-    if (stdin->_cnt > 0)
+    if (getc_unlocked(stdin) > 0)
 	return 1;
 #endif
 #endif
diff -Nur openafs-1.4.8.orig/src/shlibafsrpc/Makefile.in openafs-1.4.8/src/shlibafsrpc/Makefile.in
--- openafs-1.4.8.orig/src/shlibafsrpc/Makefile.in	2009-01-31 23:19:13.402646244 -0500
+++ openafs-1.4.8/src/shlibafsrpc/Makefile.in	2009-02-01 14:48:55.943721655 -0500
@@ -369,6 +369,10 @@
 		$(PATH_CPP)  ${SFLAGS} ${SYS}/syscall.s > syscall.ss; \
 		as -o syscall.o syscall.ss;		\
 		$(RM) syscall.ss;;				\
+	     sunamd64_*) \
+		$(PATH_CPP)  ${SFLAGS} ${SYS}/syscall.s > syscall.ss; \
+		${AS} $(SHLIB_CFLAGS) -o syscall.o syscall.ss;		\
+		$(RM) syscall.ss;;				\
 	 sgi_* | *_darwin_* ) \
                 ${CC} ${CFLAGS} -c ${SYS}/syscall.s;;          \
 	 alpha_dux?? ) \
diff -Nur openafs-1.4.8.orig/src/sys/Makefile.in openafs-1.4.8/src/sys/Makefile.in
--- openafs-1.4.8.orig/src/sys/Makefile.in	2009-01-31 23:19:12.754984733 -0500
+++ openafs-1.4.8/src/sys/Makefile.in	2009-02-01 00:07:55.066432402 -0500
@@ -91,6 +91,10 @@
 		$(PATH_CPP) -P ${SFLAGS} ${srcdir}/syscall.s > syscall.ss; \
 		as -o syscall.o syscall.ss;		\
 		$(RM) syscall.ss;;				\
+	 sunamd64_* ) \
+		$(PATH_CPP) -P ${SFLAGS} ${srcdir}/syscall.s > syscall.ss; \
+		${AS} $(SHLIB_CFLAGS) -o syscall.o syscall.ss; \
+		$(RM) syscall.ss;; \
 	 sgi_* |*_darwin* ) \
                 ${CC} ${CFLAGS} -c ${srcdir}/syscall.s;;          \
 	 alpha_osf1 | alpha_osf20 |  alpha_osf30 | alpha_osf32 | alpha_osf32c | alpha_dux?? ) \
diff -Nur openafs-1.4.8.orig/src/venus/Makefile.in openafs-1.4.8/src/venus/Makefile.in
--- openafs-1.4.8.orig/src/venus/Makefile.in	2009-01-31 23:19:12.640447302 -0500
+++ openafs-1.4.8/src/venus/Makefile.in	2009-02-01 14:51:49.040871324 -0500
@@ -143,7 +143,7 @@
 kdump: kdump.o
 	-set -x; \
 	case ${SYS_NAME} in \
-	sun4c_51 | sun4c_52 | sun4m_51 | sun4m_52 | sun4c_53 | sun4m_53 | sun4_53 | sun4_54 | sun4c_54 | sun4m_54 | sunx86_5? ) \
+	sun4c_51 | sun4c_52 | sun4m_51 | sun4m_52 | sun4c_53 | sun4m_53 | sun4_53 | sun4_54 | sun4c_54 | sun4m_54 | sunx86_5? | sunamd64_* ) \
 		${CC} -o kdump kdump.o ${TOP_LIBDIR}/libcmd.a  ${TOP_LIBDIR}/util.a /usr/lib/libkvm.a -lelf ${XLIBS} ;; \
 	sun*_5? | sun*_5?? ) \
 		${CC} -o kdump kdump.o ${TOP_LIBDIR}/libcmd.a  ${TOP_LIBDIR}/util.a  ${XLIBELFA} ${XLIBKVM} ${XLIBS} ;; \
@@ -180,7 +180,7 @@
 kdump64: 
 	-set -x; \
 	case ${SYS_NAME} in \
-	sun4x_5[789] | sun4x_510 | hp_ux11* | sunx86_5[789] | sunx86_510 )  \
+	sun4x_5[789] | sun4x_510 | hp_ux11* | sunx86_5[789] | sunx86_510 | sunamd64_511 )  \
 		$(MAKE) kdump64.o ; \
 		${CC} ${XCFLAGS64} -o kdump64 kdump64.o ${TOP_LIBDIR}/libcmd64.a ${XLIBELFA} ${XLIBKVM} ${XLIBS} ;; \
 	esac
@@ -230,7 +230,7 @@
 kdump64.o : kdump.c ${INCLS} AFS_component_version_number.c
 	-set -x; \
 	case ${SYS_NAME} in \
-	sun4x_5[789] | sun4x_510 | hp_ux11* | sunx86_5[789] | sunx86_510 ) \
+	sun4x_5[789] | sun4x_510 | hp_ux11* | sunx86_5[789] | sunx86_510 | sunamd64_511 ) \
 		${CC} ${KERN_DBG} ${KERN_OPTMZ} -I${TOP_SRCDIR} -I${TOP_INCDIR}/afs \
 		-I${TOP_OBJDIR}/src/config -I${TOP_OBJDIR}/src -I${TOP_INCDIR} ${XCFLAGS64} \
 		-o kdump64.o -c ${srcdir}/kdump.c ;; \
diff -Nur openafs-1.4.8.orig/src/venus/up.c openafs-1.4.8/src/venus/up.c
--- openafs-1.4.8.orig/src/venus/up.c	2009-01-31 23:19:12.639641435 -0500
+++ openafs-1.4.8/src/venus/up.c	2009-02-02 17:45:32.981695135 -0500
@@ -51,11 +51,6 @@
 
 #define MAXACL 400
 
-#if !defined(AFS_LINUX20_ENV) && !defined(AFS_DARWIN_ENV) && !defined(AFS_XBSD_ENV)
-extern sys_nerr;
-extern char *sys_errlist[];
-#endif
-
 short verbose = 0;
 short renameTargets = 0;
 short oneLevel = 0;
@@ -251,6 +246,7 @@
 	 */
 	int f1, f2, n;
 	char buf[4096];		/* Must be bigger than sizeof (*head) */
+	char errmsg[512];
 	struct timeval tv[2];
 	char tmpfile[MAXPATHLEN], newName[MAXPATHLEN];
 
@@ -288,22 +284,16 @@
 	    /* open file1 for input */
 	    f1 = open(file1, O_RDONLY);
 	    if (f1 < 0) {
-		fprintf(stderr, "Unable to open input file %s, ", file1);
-		if (errno >= sys_nerr)
-		    fprintf(stderr, "error code = %d\n", errno);
-		else
-		    fprintf(stderr, "%s\n", sys_errlist[errno]);
+		sprintf(errmsg, "Unable to open input file %s", file1);
+		perror(errmsg);
 		return 1;
 	    }
 
 	    /* open temporary output file */
 	    f2 = open(tmpfile, (O_WRONLY | O_CREAT | O_TRUNC), s1.st_mode);
 	    if (f2 < 0) {
-		fprintf(stderr, "Unable to open output file %s, ", tmpfile);
-		if (errno >= sys_nerr)
-		    fprintf(stderr, "error code = %d\n", errno);
-		else
-		    fprintf(stderr, "%s\n", sys_errlist[errno]);
+		sprintf(errmsg, "Unable to open input file %s", tmpfile);
+		perror(errmsg);
 		fflush(stdout);
 		close(f1);
 		return 1;
diff -Nur openafs-1.4.8.orig/src/vol/AFS_component_version_number.c openafs-1.4.8/src/vol/AFS_component_version_number.c
--- openafs-1.4.8.orig/src/vol/AFS_component_version_number.c	1969-12-31 19:00:00.000000000 -0500
+++ openafs-1.4.8/src/vol/AFS_component_version_number.c	2009-02-01 00:17:01.643256450 -0500
@@ -0,0 +1,2 @@
+char cml_version_number[]="@(#) OpenAFS 1.4.8 built  2009-02-01 ";
+char* AFSVersion = "openafs 1.4.8"; 
diff -Nur openafs-1.4.8.orig/src/vol/clone.c openafs-1.4.8/src/vol/clone.c
--- openafs-1.4.8.orig/src/vol/clone.c	2009-01-31 23:19:12.452498923 -0500
+++ openafs-1.4.8/src/vol/clone.c	2009-02-01 14:56:14.407177762 -0500
@@ -37,6 +37,7 @@
 #else
 #include <sys/file.h>
 #include <sys/time.h>
+#include <unistd.h>
 #endif
 #include <string.h>
 #include <errno.h>
diff -Nur openafs-1.4.8.orig/src/vol/ihandle.c openafs-1.4.8/src/vol/ihandle.c
--- openafs-1.4.8.orig/src/vol/ihandle.c	2009-01-31 23:19:12.454352288 -0500
+++ openafs-1.4.8/src/vol/ihandle.c	2009-02-01 00:17:01.650848372 -0500
@@ -27,6 +27,7 @@
 #include <sys/file.h>
 #include <unistd.h>
 #include <sys/stat.h>
+#include <fcntl.h>
 #if defined(AFS_SUN5_ENV) || defined(AFS_NBSD_ENV)
 #include <sys/fcntl.h>
 #include <sys/resource.h>
diff -Nur openafs-1.4.8.orig/src/vol/partition.c openafs-1.4.8/src/vol/partition.c
--- openafs-1.4.8.orig/src/vol/partition.c	2009-01-31 23:19:12.452726021 -0500
+++ openafs-1.4.8/src/vol/partition.c	2009-02-01 00:17:01.655300950 -0500
@@ -32,6 +32,7 @@
 #else
 #include <sys/param.h>
 #include <sys/types.h>
+#include <fcntl.h>
 
 #if AFS_HAVE_STATVFS || AFS_HAVE_STATVFS64
 #include <sys/statvfs.h>
diff -Nur openafs-1.4.8.orig/src/vol/vol-info.c openafs-1.4.8/src/vol/vol-info.c
--- openafs-1.4.8.orig/src/vol/vol-info.c	2009-01-31 23:19:12.454274485 -0500
+++ openafs-1.4.8/src/vol/vol-info.c	2009-02-01 00:17:01.655907080 -0500
@@ -33,6 +33,7 @@
 #include <sys/param.h>
 #include <sys/file.h>
 #include <sys/time.h>
+#include <fcntl.h>
 #endif
 #include <afs/cmd.h>
 
diff -Nur openafs-1.4.8.orig/src/vol/volume.c openafs-1.4.8/src/vol/volume.c
--- openafs-1.4.8.orig/src/vol/volume.c	2009-01-31 23:19:12.455135031 -0500
+++ openafs-1.4.8/src/vol/volume.c	2009-02-01 00:17:01.666819950 -0500
@@ -58,6 +58,7 @@
 #include <fcntl.h>
 #else
 #include <sys/file.h>
+#include <fcntl.h>
 #endif
 #include <dirent.h>
 #ifdef	AFS_AIX_ENV
diff -Nur openafs-1.4.8.orig/src/vol/vutil.c openafs-1.4.8/src/vol/vutil.c
--- openafs-1.4.8.orig/src/vol/vutil.c	2009-01-31 23:19:12.452429921 -0500
+++ openafs-1.4.8/src/vol/vutil.c	2009-02-01 14:24:32.489933106 -0500
@@ -30,6 +30,7 @@
 #include <sys/time.h>
 #include <sys/file.h>
 #include <unistd.h>
+#include <fcntl.h>
 #endif
 #include <sys/stat.h>
 #ifdef AFS_PTHREAD_ENV
