--- src/yaws_server.erl.old	2009-08-01 18:57:30.097138117 -0400
+++ src/yaws_server.erl	2009-08-01 23:26:52.046145929 -0400
@@ -2095,10 +2095,16 @@
     {P, Q} = yaws:split_at(DecPath, $?),
     LocPath = yaws_api:format_partial_url(URL, get(sc)),
     Loc = if
-              Mode == append, Q == [] ->
-                  ["Location: ", LocPath, P, "\r\n"];
-              Mode == append, Q /= [] ->
-                  ["Location: ", LocPath, P, "?", Q, "\r\n"];
+              Mode == append ->
+                  Newpath = filename:join([URL#url.path ++ P]),
+                  NLocPath = yaws_api:format_partial_url(
+                               URL#url{path = Newpath}, get(sc)),
+                  case Q of
+                      [] ->
+                          ["Location: ", NLocPath, "\r\n"];
+                      _Q ->
+                          ["Location: ", NLocPath, "?", Q, "\r\n"]
+                  end;
               Mode == noappend,Q == [] ->
                   ["Location: ", LocPath, "\r\n"];
               Mode == noappend,Q /= [] ->
