--- src/mime.types.old	2009-08-14 21:25:56.964295026 -0400
+++ src/mime.types	2009-08-14 21:26:17.235753891 -0400
@@ -51,7 +51,7 @@
 application/news-transmission
 application/ocsp-request
 application/ocsp-response
-application/octet-stream	bin dms lha lzh exe class so dll jar cab
+application/octet-stream	bin dms lha lzh exe class so dll cab
 application/oda			oda
 application/parityfec
 application/pdf			pdf
@@ -301,6 +301,10 @@
 application/x-gtar		gtar
 application/x-gzip		gz tgz
 application/x-hdf		hdf
+application/x-java-jnlp-file	jnlp
+application/vnd.rim.cod        cod
+text/vnd.sun.j2me.app-descriptor        jad
+application/java-archive	jar
 application/x-javascript	js
 application/x-kword		kwd kwt
 application/x-kspread		ksp
